ami-recycle-bin/main.tf

```diff
@@ -0,0 +1,144 @@
+provider "aws" {
+ region = "us-west-2"
+}
+
+terraform {
+ required_providers {
+   aws = {
+     source  = "hashicorp/aws"
+     version = "5.65.0"
+   }
+ }
+}
+
+resource "aws_cloudwatch_event_rule" "event_rule" {
+  name                = "invoke-lambda-daily"
+  description         = "Invoke a Lambda function every day"
+  schedule_expression = "rate(1 day)"
+}
+
+resource "aws_cloudwatch_event_target" "event_target" {
+  rule      = aws_cloudwatch_event_rule.event_rule.name
+  target_id = "InvokeLambda"
+  arn       = aws_lambda_function.lambda_function.arn
+}
+
+resource "aws_lambda_permission" "lambda_permission" {
+  statement_id  = "AllowExecutionFromEventBridge"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.lambda_function.function_name
+  principal     = "events.amazonaws.com"
+  source_arn    = aws_cloudwatch_event_rule.event_rule.arn
+}
+
+resource "aws_cloudwatch_log_group" "example" {
+  name              = "/aws/lambda/${aws_lambda_function.lambda_function.function_name}"
+  retention_in_days = var.cw_retention_period
+}
+
+resource "aws_lambda_function" "lambda_function" {
+  filename         = "src/ami-recycle-lambda.zip"
+  function_name    = "ami-recycle-lambda"
+  role             = aws_iam_role.lambda_role.arn
+  handler          = "ami-recycle-lambda.lambda_handler"
+  runtime          = "python3.9"
+  # Add environment variable block here
+  environment {
+    variables = {
+      "RECYCLE_BIN_TAG_KEY" = var.resource_tag_key
+      "RECYCLE_BIN_TAG_VALUE" = var.resource_tag_value
+      "RBIN_RETENTION_PERIOD_VALUE" = var.rbin_retention_period_value
+      "RBIN_RETENTION_PERIOD_UNIT" = var.rbin_retention_period_unit
+    }
+  }
+  timeout          = 300
```

Comment:
Is such a long timeout really necessary?

Reply:
> Is such a long timeout really necessary?

If the list of expired AMIs is long, the maximum timeout will be helpful to ensure function does not timeout before the expired AMIs and its snapshot is moved to Recycle Bin. However since this is a test pattern, updated the `timeout` parameter as a configurable variable and set the default timeout to `15 seconds`